metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -319,20 +322,37 @@ final class BuildTargets(
     }
   }
 
+  /**
+   *  Match build target by source jar name if source was saved in `.metals/readonly/dependecies/${source-jar-name}`
+   */
+  private def inferByJarDirectoryName(
+      source: AbsolutePath
+  ): Option[BuildTargetIdentifier] = {
+    if (source.isInDependenciesDirectory(workspace)) {
+      val jarName =
+        source
+          .toRelative(workspace.resolve(Directories.dependencies))
+          .toNIO
+          .getName(0)
+          .filename
+
+      inverseDependencySources
+        .find { case (path, _) =>
+          path.filename == jarName
+        }
+        .flatMap { case (_, ids) => ids.headOption }
```

Comment:
I must be losing my mind or something but I swear I already commented this, but now it's gone 🤔 anways, probably just style choice, but you could also do

```scala
        .collectFirst {
          case (path, ids) if path.filename == jarName && ids.nonEmpty =>
            ids.head
        }
```
To avoid iterating twice.

I wish we had a `nonEmptySet` we could use here since realistically `ids` will never be empty.

Reply:
Thx! Replaced by `collectFirst`